src/user/components/DownloadObservations.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { API_ROOT } from "../../app/app-helpers";
+import axios from "axios";
+
+export default class DownloadObservations extends React.Component {
+  state = {
+    isLoading: false,
+    errorMessage: "",
+    csvData: []
+  };
+
+  linkRef = React.createRef();
+
+  download = async () => {
+    this.setState({ errorMessage: `` });
+    this.setState({ isLoading: true });
+
+    try {
+      const result = await axios.post(
+        `${API_ROOT}/getAllObservations`,
+        {},
+        {
+          withCredentials: true,
+          headers: {
+            "Content-Type": "application/json"
+          }
+        }
+      );
+
+      // If replacing a previously generated file, revoke the object URL to avoid memory leaks.
+      if (this.state.csvFile !== null) {
+        window.URL.revokeObjectURL(this.state.csvData);
+      }
+
+      this.setState({
+        csvData: result.data
+      });
+
+      const href = window.URL.createObjectURL(
+        new Blob([this.state.csvData], {
+          type: "text/csv"
+        })
+      );
+      this.linkRef.current.download = "my_observations.csv";
+      this.linkRef.current.href = href;
+      this.linkRef.current.click();
+      this.linkRef.current.href = "";
+    } catch (error) {
+      this.setState({ errorMessage: error.response.data });
```

Comment:
`Something went wrong... <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN"> <title>500 Internal Server Error</title> <h1>Internal Server Error</h1> <p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p> `

^ this error is returned when user has 0 obs and they click. button should be hidden if they dont have obs,